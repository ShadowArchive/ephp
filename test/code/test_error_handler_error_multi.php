<?php

function handle_error($errno, $errstr, $errfile, $errline, $errcontext=array()) {
    static $a = 0;
    $errfile = basename($errfile);
    print "$errfile:$errline [$errno] $errstr\n";
    set_error_handler("handle_error2");
    trigger_error("whoops!", E_USER_ERROR);
}

function handle_error2($errno, $errstr, $errfile, $errline, $errcontext=array()) {
    print "SECOND($errno): $errstr\n";
}

set_error_handler("handle_error");

# generate a warning...
sqrt("a");

trigger_error("Noooo!", E_ERROR);

print "NO OK\n";
