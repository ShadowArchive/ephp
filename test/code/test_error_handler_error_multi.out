test_error_handler_error_multi.php:18 [2] sqrt() expects parameter 1 to be double, string given
SECOND(256): whoops!
SECOND(2): Invalid error type specified
NO OK
